### ISSUE TITLE
AggregateExpr `Avg`  removeOffset bug

### ISSUE THREAD

**Issue by sundy-li** [jacksontj/promxy#275]
From the logs

```
DEBU[2020-03-04T11:03:41+08:00] AggregateExpr avg(call_reranker_latency_90{job="popular"} offset 1d) avg
DEBU[2020-03-04T11:03:41+08:00] AggregateExpr count(call_reranker_latency_90{job="popular"} offset 1d) count
DEBU[2020-03-04T11:03:41+08:00] AggregateExpr sum(call_reranker_latency_90{job="popular"}) sum
```

We can see that avg was transformed to `sum/count` Binary OP, yet the count/sum will do the `removeOffset` for the same `expr`, so that's the problem.

I want to contribute a PR to fix that, but the `promql.Expr` didn't expose a method like `Clone`,  so it's not an easy job.

**Comment by sundy-li** [comment]
BTW,  the code seems a little compicated,  I didn't know why we rewrite this `n.Op = promql.ItemSum` in https://github.com/jacksontj/promxy/blob/master/pkg/proxystorage/proxy.go#L335

**Comment by sundy-li** [comment]
I found a tricky way to clone the `Expr`.

**Comment by jacksontj** [comment]
That fix looks good, I have also opened up #277 to add tests to cover this case, it seems that avg() was the only one with the issue.
